pysat/instruments/methods/sw.py

```diff
@@ -511,3 +511,74 @@ def calc_daily_Ap(ap_inst, ap_name='3hr_ap', daily_name='Ap',
     ap_inst.meta.__setitem__(daily_name, meta_dict)
 
     return
+
+def convert_ap_to_kp(ap_data, fill_val=-1, ap_name='ap'):
+    """ Convert Ap into Kp
+
+    Parameters
+    ----------
+    ap_data : array-like
+        Array-like object containing Ap data
+    fill_val : int, float, NoneType
+        Fill value for the data set (default=-1)
+    ap_name : str
+        Name of the input ap
+
+    Returns
+    -------
+    kp_data : array-like
+        Array-like object containing Kp data
+    meta : Metadata
+        Metadata object containing information about transformed data
+
+    """
+    
+    # Ap are keys, Kp returned as double (N- = N.6667, N+=N.3333333)
+    one_third = 1.0 / 3.0
+    two_third = 2.0 / 3.0
+    ap_to_kp = {0: 0, 2: one_third, 3: two_third, 4: 1, 5: 1.0+one_third,
+                6: 1.0+two_third, 7: 2, 9: 2.0+one_third, 12: 2.0+two_third,
+                15: 3, 18: 3.0+one_third, 22: 3.0+two_third, 27: 4,
+                32: 4.0+one_third, 39: 4.0+two_third, 48: 5, 56: 5.0+one_third,
+                67: 5.0+two_third, 80: 6, 94: 6.0+one_third, 111: 6.0+two_third,
+                132: 7, 154: 7.0+one_third, 179: 7.0+two_third, 207: 8,
+                236: 8.0+one_third, 300: 8.0+two_third, 400: 9}
+    ap_keys = sorted(list(ap_to_kp.keys()))
+
+    # If the ap falls between two Kp indices, assign it to the lower Kp value
+    def round_ap(ap_in, fill_val=fill_val):
+        """ Round an ap value to the nearest Kp value
+        """
+        if not np.isfinite(ap_in):
+            return fill_val
+        
+        i = 0
```

Comment:
Starting at line 555 (here) down to the return aren't tested with the unit tests.

Reply:
Added three more unit tests for these cases.